android/java/org/chromium/chrome/browser/crypto_wallet/activities/NetworkSelectorActivity.java

```diff
@@ -58,27 +72,48 @@ public void finishNativeInitialization() {
         initState();
     }
 
+    @SuppressLint("NotifyDataSetChanged")
+    @Override
+    public void onActivityResult(int requestCode, int resultCode, Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+        if (resultCode != RESULT_OK) return;
+
+        BraveActivity activity = BraveActivity.getBraveActivity();
+        if (activity != null) {
+            activity.getWalletModel().getCryptoModel().getNetworkModel().init();
```

Comment:
Not sure if we can listen to network add events via observers like `JsonRpcServiceObserver` if yes then prefer to trigger `init` inside model via the observer otherwise, create a new method `refershNetworks` and use it here. `init` can change anytime so prefer to avoid calling it directly outside the `domain`. (maybe we could change it to default scope later)

Reply:
I haven't found a way that listens for add events, unfortunately. I'm falling back to the second solution and creating a new `refreshNetworks` method in the model.